src/ORM/RelatedData/StandardRelatedDataService.php

```diff
@@ -314,8 +314,10 @@ private function fetchReverseHasOneResults(DataObject $record, string $class, st
         $where = sprintf('"%s" = %u', $componentIDField, $record->ID);
 
         // Polymorphic
-        if ($componentIDField === 'ParentID' && isset($dbFields['ParentClass'])) {
-            $where .= sprintf(' AND "ParentClass" = %s', $this->prepareClassNameLiteral(get_class($record)));
+        foreach (['Parent', 'Owner'] as $poly) {
+            if ($componentIDField === $poly . 'ID' && isset($dbFields[$poly . 'Class'])) {
+                $where .= sprintf(' AND "' . $poly . 'Class" = %s', $this->prepareClassNameLiteral(get_class($record)));
+            }
```

Comment:
In the case of `Owner` for links, we also need to check the relation....

It's probably worth doing this correctly - there's no guarantee that `Parent` nor `Owner` will be polymoprhic for any given record, but here we're pretending they will _always_ be polymorphic.

We should instead just check if `$component` is polymorphic (if `DataObject::getSchema()->hasOneComponent($class, $component) === DataObject::class` is true, it's polymorphic), and if it's multi-relational (if `DataObject::getSchema()->hasOneComponentHandlesMultipleRelations($class, $component)` is true, it's multi-relational).
It doesn't matter if it's named `Parent` or `Owner` or something else.

Reply:
Thinking about it again we probably don't need the multi-relational part - but we should make it generic for all polymorphic relations.

Reply:
Updated - note there's an existing polymorphic test in RelatedDataServiceTest.php which is green after this code change